### ISSUE TITLE
FreeBSD: Invalid value for ai_flags

### ISSUE THREAD

**Issue by ghost** [AGWA/titus#2]
FreeBSD does not support AI_V4MAPPED. Titus compiles, but cannot bind to a socket.

**Comment by AGWA** [comment]
While I've got your attention, @feld, I noticed that FreeBSD now has a way to disable debugging a process (https://svnweb.freebsd.org/base?view=revision&revision=277322).  Has this made its way into any releases yet?

**Comment by ghost** [comment]
FreeBSD 10.2-RELEASE has that commit

**Comment by AGWA** [comment]
@feld, could you give the current `master` branch a try? It eliminates the dependence on `AI_V4MAPPED` and uses `procctl(2)` with `PROC_TRACE_CTL` to prevent tracing, which should make titus as secure on FreeBSD as it is on Linux.

If it works, I'll cut a release.

**Comment by ghost** [comment]
I think there's something odd yet about the socket binding but it works. Ex, listen on an ipv4 address and netstat looks like this:

```
tcp46      0      0 *.4443                 *.*                    LISTEN
```

which appears that it's listening on all addresses, but it only lets me connect to what is defined in the titus.conf.

**Comment by ghost** [comment]
Then again, all of the tcp46 entries in netstat look like that. Maybe there's a different bug I need to complain about to FreeBSD developers

**Comment by AGWA** [comment]
Oh, are you using the `local-address` option in titus?  That doesn't actually change what address titus listens on.  titus always listens on all addresses, and uses `local-address` to match incoming connections to a vhost.  Ideally, titus would only listen on the addresses for which vhosts are defined, but it is surprisingly tricky to efficiently listen on multiple addresses from a preforked server.

**Comment by ghost** [comment]
That explains it. Thanks for the clarification!

I think this is ready to be cut into a release. The only modification I have to make for FreeBSD is to alter the Makefile to install the man page into $(MANPREFIX)/man/man8

**Comment by AGWA** [comment]
@feld I just released titus 0.3.  I fixed the Makefile to properly install the man page to the man8 sub-directory, so all you should need to do is set `MANDIR` to `$(MANPREFIX)/man`and the Makefile will do what you want.

**Comment by ghost** [comment]
and I've committed this to FreeBSD ports

Thanks!
